javav2/example_code/s3/src/main/java/com/example/s3/express/S3DirectoriesScenario.java

```diff
@@ -0,0 +1,578 @@
+        }
+        waitForInputToContinue(scanner);
+    }
+
+    private static String createSessionAddObjects() {
+        logger.info("""    
+            5. Create an object and copy it over.
+            We'll create a basic object consisting of some text and upload it to the 
+            normal bucket. 
+            Next we'll copy the object into the Directory bucket using the regular client. 
+            This works fine because copy operations are not restricted for Directory buckets.
+            """);
+        waitForInputToContinue(scanner);
+
+        String bucketObject = "basic-text-object.txt";
+        try {
+            s3DirectoriesActions.putObjectAsync(mS3RegularClient, mregularBucketName, bucketObject, "Look Ma, I'm a bucket!").join();
+            s3DirectoriesActions.createSessionAsync(mS3ExpressClient, mdirectoryBucketName).join();
+
+            // Copy the object to the destination S3 bucket.
+            s3DirectoriesActions.copyObjectAsync(mS3ExpressClient, mregularBucketName, bucketObject, mdirectoryBucketName, bucketObject).join();
+        } catch (CompletionException ce) {
+            Throwable cause = ce.getCause();
+            if (cause instanceof S3Exception) {
+                logger.error("S3Exception occurred: {}", cause.getMessage(), ce);
+            } else {
+                logger.error("An unexpected error occurred: {}", cause.getMessage(), ce);
+            }
+        }
+        logger.info(""" 
+            It worked! It's important to remember the user permissions when interacting with 
+            Directory buckets. Instead of validating permissions on every call as 
+            normal buckets do, Directory buckets utilize the user credentials and session 
+            token to validate. This allows for much faster connection speeds on every call. 
+            For single calls, this is low, but for many concurrent calls 
+            this adds up to a lot of time saved.
+            """);
+        waitForInputToContinue(scanner);
+        return bucketObject;
+    }
+
+    /**
+     * Creates VPC users for the S3 Express One Zone scenario.
+     * <p>
+     * This method performs the following steps:
+     * <ol>
+     *     <li>Optionally creates a new VPC and VPC Endpoint if the application is running in an EC2 instance in the same Availability Zone as the Directory buckets.</li>
+     *     <li>Creates two IAM users: one with S3 Express One Zone permissions and one without.</li>
+     * </ol>
+     *
+     * @return a {@link UserNames} object containing the names of the created IAM users
+     */
+    public static UserNames createVpcUsers() {
+        /*
+        Optionally create a VPC.
+        Create two IAM users, one with S3 Express One Zone permissions and one without.
+        */
+        logger.info("""
+            1. First, we'll set up a new VPC and VPC Endpoint if this program is running in an EC2 instance in the same AZ as your\s
```

Comment:
Again to the issue of running this Basics on an EC2 instance. I doubt this will happen. 

Also, I don't think users will know how to answer "Are you running this in an EC2 instance located in the same AZ as your intended Directory buckets?" since they don't know what the options for AZ zones are yet, even if they know that they are running their EC2 instance in specific AZ.

Reply:
Lets look at changing some of the text